Modix.Bot/ModixBot.cs

```diff
@@ -31,7 +32,9 @@ public sealed class ModixBot : BackgroundService
         private readonly ModixConfig _config;
         private readonly DiscordSerilogAdapter _serilogAdapter;
         private readonly IApplicationLifetime _applicationLifetime;
+        private readonly CommandErrorHandler _commandErrorHandler;
         private IServiceScope _scope;
+        private readonly ConcurrentDictionary<ICommandContext, IServiceScope> _commandScopes = new ConcurrentDictionary<ICommandContext, IServiceScope>();
```

Comment:
Do we need to use `ConcurrentDictionary` if we only ever add a brand new instance of `ICommandContext`?

Reply:
Not sure how regular dictionaries handle multithreaded adds of different keys. I thought it wasn't threadsafe and could corrupt the internal collection

Reply:
Sticking with a `ConcurrentDictionary` is fine, it definitely seems like the safer approach.